models/status.go

```
@@ -219,7 +219,9 @@ func newCommitStatus(sess *xorm.Session, opts NewCommitStatusOptions) error {
 	}
 	has, err := sess.Desc("index").Limit(1).Get(lastCommitStatus)
 	if err != nil {
-		sess.Rollback()
+		if err := sess.Rollback(); err != nil {
+			return err
```

IMHO this error should only be logged and the method should return the initial error.
Same for L237